src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1009,4 +1010,61 @@ public void setSecurityAnalysisServerBaseUri(String securityAnalysisServerBaseUr
     public void setActionsServerBaseUri(String actionsServerBaseUri) {
         this.actionsServerBaseUri = actionsServerBaseUri;
     }
+
+    public Mono<Void> stopSecurityAnalysis(String studyName, String userId) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
+
+            String receiver;
+            try {
+                receiver = URLEncoder.encode(objectMapper.writeValueAsString(new Receiver(studyName, userId)), StandardCharsets.UTF_8);
+            } catch (JsonProcessingException e) {
+                throw new UncheckedIOException(e);
+            }
+            return Mono.justOrEmpty(resultUuid).flatMap(uuid -> {
+                String path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/results/{resultUuid}/stop")
+                        .queryParam(RECEIVER, receiver)
+                        .buildAndExpand(resultUuid)
+                        .toUriString();
+                return webClient
+                        .put()
+                        .uri(securityAnalysisServerBaseUri + path)
+                        .retrieve()
+                        .bodyToMono(Void.class);
+            });
+        });
+    }
+
+    @Bean
+    public Consumer<Flux<Message<String>>> consumeSaStopped() {
+        return f -> f.log(CATEGORY_BROKER_INPUT, Level.FINE).flatMap(message -> {
+            UUID resultUuid = UUID.fromString(message.getHeaders().get("resultUuid", String.class));
+            String receiver = message.getHeaders().get(RECEIVER, String.class);
+            if (receiver != null) {
+                Receiver receiverObj;
+                try {
+                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), Receiver.class);
+
+                    LOGGER.info("Security analysis stopped for study '{}' and user '{}'",
+                            resultUuid, receiverObj.getStudyName(), receiverObj.getUserId());
+
+                    // delete security analysis result in database
+                    return studyRepository.updateSecurityAnalysisResultUuid(receiverObj.getStudyName(), receiverObj.getUserId(), null)
+                            .then(Mono.fromCallable(() -> {
+                                // send notification for stopped computation
+                                emitStudyChanged(receiverObj.getStudyName(), UPDATE_TYPE_SECURITY_ANALYSIS_STATUS);
+                                return null;
+                            }));
+                } catch (JsonProcessingException e) {
+                    LOGGER.error(e.toString());
+                }
+            }
+            return Mono.empty();
+        })
+                .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
```

Comment:
this creates ugly logs no ? The message is printed twice. Instead use something like
``` 
LOGGER.error("Unkown error in consumming sastopped message" , throwable))
```
(untested, please test it)